DOCO-1055 FES Rendering Modes (#8529)
pantheon-systems/documentation: source/content/guides/decoupled/overview/03-site-options.md

| WordPress  |   5.9 or higher |

## Rendering Options

This section provides a comparison of static site generation (SSG), server side rendering (SSR), and incremental static regeneration (ISR) modes, and some common use cases for each.

Refer to [Decoupled Architectures: What Computer Assembles the Websites](https://pantheon.io/blog/decoupled-architectures-what-computer-assembles-websites) for more information. You can also check out this [Decoupled Architecture video](https://www.youtube.com/watch?v=dF39cXW3IqY) for a demonstration of the different approaches each mode takes.

### What Is Static Site Generation (SSG)?

> SSG compiles and renders your website at build time. SSG output consists of:

Reviewer: ```suggestion
SSG compiles and renders your entire website at build time. SSG output consists of:
```

- **Static files:** HTML
- **Assets:** JavaScript, CSS, etc.

#### Common SSG Use Cases

> 

Reviewer: ```suggestion
- Sites where uptime is critical.  Since SSG sites have all their content dependencies met at build time, a failure by a dependent content source will only block a content update.  If a CMS is down when a SSR site needs it's content to render a page the SSR site will not be able to return a result. 
```

- Pages with content that is the same for all users

> - Pages that can be publicly cached

Reviewer: - Websites with infrequent content changes/edits benefit from SSG since the cost for longer build time renders is not a major penalty.

> - Pages that can be publicly cached

Reviewer: ```suggestion
- Websites with regular content changes/edits benefit from SSR since they do not need to compile the entire website at once like they would in SSG.
```

> - Pages that do not require data from an outside source

Reviewer: ```suggestion
-  Pages that do not require back and forth processing with a runtime.  
```

Note data can be fetched from outside sources at build time.

### What Is Server Side Rendering (SSR)?

> SSR uses a web server to display your site instead of a browser. The server prepares your HTML file with user-specific data and sends this to the user's browser. The browser interprets the content and displays a fully rendered HTML page without waiting for CSS or JavaScript files to load.

Reviewer: > Where SSG builds the HTML for a webpage at Build Time.  SSR builds the HTML for a webpage at request time.  As the request comes into a node.js runtime it builds a page much like a Drupal or Wordpress instance might.

I'd delete "The browser interprets the content and displays a fully rendered HTML page without waiting for CSS or JavaScript files to load."  That is the case for both SSR vs SSG

> SSR uses a web server to display your site instead of a browser. The server prepares your HTML file with user-specific data and sends this to the user's browser. The browser interprets the content and displays a fully rendered HTML page without waiting for CSS or JavaScript files to load.

Reviewer: ```suggestion
SSR uses a web server running node.js to render individual pages on the fly as they are requested.  This is different from SSG which renders the entire site at build time. 
```

### Common SSR Use Cases

- Pages that are user specific
- Pages that require authentication

> - Pages that require data fetched at request time

Reviewer: ```suggestion
- Pages that do not require back and forth processing with a runtime.  
- Websites with regular content changes/edits benefit from SSR since they do not need to compile the entire website at once like they would in SSG.

```

### What Is Incremental Static Regeneration (ISR)?

<Alert title="Note"  type="info" >

Pantheon does not support Incremental Static Regeneration (ISR).

</Alert>

ISR allows you to regenerate static pages during runtime without having to rebuild your site. Refer to [Next.js ISR](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration) for more information.

### Key Differences

> - SSG compiles and renders your website at build time.

Reviewer: ```suggestion
- SSG compiles and renders your entire website at build time.
```

> - SSR uses a web server to display your site instead of a browser. SSR pages fetch data at the time the page is requested.

Reviewer: ```suggestion
- SSR uses a web server to render individual webpages on demand as opposed to SSG which renders entire websites at build time.
```

> - ISR is not supported on the Pantheon platform. ISR pages are SSG pages that refetch data after a given time has expired.

Reviewer: ```suggestion
- ISR is not supported on the Pantheon platform.  Like SSG, ISR renders individual pages on the fly.  However, ISR stores the results in an internal cache.  This cache is solving problems which are already better solved by CDNs.  Pantheon ships with a Global CDN.  The combination of SSR + CDN solves the same problem as ISR in a more robust way.
```
